tracer/src/Datadog.Trace/Ci/Configuration/CIVisibilitySettings.cs

```diff
@@ -11,15 +12,60 @@ internal class CIVisibilitySettings
     {
         public CIVisibilitySettings(IConfigurationSource source)
         {
-            Enabled = source?.GetBool(ConfigurationKeys.CIVisibilityEnabled) ?? false;
-            if (Enabled)
-            {
-                TracerSettings = new TracerSettings(source);
-            }
+            Enabled = source?.GetBool(ConfigurationKeys.CIVisibility.Enabled) ?? false;
+            Agentless = source?.GetBool(ConfigurationKeys.CIVisibility.AgentlessEnabled) ?? false;
+            ApiKey = source?.GetString(ConfigurationKeys.ApiKey);
+            Site = source?.GetString(ConfigurationKeys.Site) ?? "datadoghq.com";
+
+            // By default intake payloads has a 5MB limit
+            MaximumAgentlessPayloadSize = 5 * 1024 * 1024;
+
+            ProxyHttps = source?.GetString(ConfigurationKeys.Proxy.ProxyHttps);
+            var proxyNoProxy = source?.GetString(ConfigurationKeys.Proxy.ProxyNoProxy) ?? string.Empty;
+            ProxyNoProxy = proxyNoProxy.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+
+            TracerSettings = new TracerSettings(source);
         }
 
+        /// <summary>
+        /// Gets a value indicating whether the CI Visibility mode was enabled by configuration
+        /// </summary>
         public bool Enabled { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether the Agentless writer is going to be used.
+        /// </summary>
+        public bool Agentless { get; }
+
+        /// <summary>
+        /// Gets the Api Key to use in Agentless mode
+        /// Note: This enables the Agentless mode
```

Comment:
nit: You need to set `Agentless` to `true` as well right, so it's a _requirement_ but it doesn't _enable_ agentless?

Reply:
ohh, right, the first approach was using only this var. I've removed the note in: [5b2e33c](https://github.com/DataDog/dd-trace-dotnet/pull/2402/commits/5b2e33c4fef2ca12412446d9ebc4637f60f24029)